src/query/storages/common/table-meta/src/meta/v5/segment.rs

```diff
@@ -0,0 +1,205 @@
+        Self::LeafSegment(LeafSegmentInfo {
+            format_version: LeafSegmentInfo::VERSION,
+            blocks,
+            summary,
+        })
+    }
+
+    #[allow(unused)]
+    pub fn new_internal(segments: Vec<Arc<Location>>, summary: Statistics) -> Self {
+        Self::InternalSegment(InternalSegmentInfo {
+            format_version: SegmentInfo::VERSION,
+            son_segments: segments,
+            summary,
+        })
+    }
+
+    // Total block bytes of this segment.
+    #[allow(unused)]
+    pub fn total_bytes(&self) -> u64 {
+        match self {
+            Self::LeafSegment(leaf_segment_info) => {
+                leaf_segment_info.summary.uncompressed_byte_size
+            }
+            Self::InternalSegment(internal_segment_info) => {
+                internal_segment_info.summary.uncompressed_byte_size
+            }
+        }
+    }
+
+    /// Serializes the Segment struct to a byte vector.
+    ///
+    /// The byte vector contains the format version, encoding, compression, and compressed block data and
+    /// summary data. The encoding and compression are set to default values. The block data and summary
+    /// data are encoded and compressed, respectively.
+    ///
+    /// # Returns
+    ///
+    /// A Result containing the serialized Segment data as a byte vector. If any errors occur during
+    /// encoding, compression, or writing to the byte vector, an error will be returned.
+    pub fn to_bytes(&self) -> Result<Vec<u8>> {
+        match self {
+            Self::LeafSegment(leaf_segment_info) => leaf_segment_info.to_bytes(),
+            Self::InternalSegment(internal_segment_info) => internal_segment_info.to_bytes(),
+        }
+    }
+
+    pub fn from_slice(bytes: &[u8]) -> Result<Self> {
+        let segment;
+        let mut cursor = Cursor::new(bytes);
+        let SegmentHeader {
+            version,
+            encoding,
+            compression,
+            // maybe we need to modify as blocks_or_segments_size
+            blocks_size,
+            summary_size,
+        } = decode_segment_header(&mut cursor)?;
+
+        if version == LeafSegmentInfo::VERSION {
+            let blocks: Vec<Arc<BlockMeta>> =
+                read_and_deserialize(&mut cursor, blocks_size, &encoding, &compression)?;
+            let summary: Statistics =
+                read_and_deserialize(&mut cursor, summary_size, &encoding, &compression)?;
+
+            segment = Self::new_leaf(blocks, summary);
+        } else if version == Self::VERSION {
+            let son_segments: Vec<Arc<Location>> =
+                read_and_deserialize(&mut cursor, blocks_size, &encoding, &compression)?;
+            let summary: Statistics =
+                read_and_deserialize(&mut cursor, summary_size, &encoding, &compression)?;
+
+            segment = Self::new_internal(son_segments, summary);
+        } else {
+            panic!("read bad version for v5")
```

Comment:
I might understand incorrectly, but how about making the `pub enum SegmentInfo` `Serialize` and `Deserialize`

Reply:
Just use to_bytes and from_slice, is that incorrect?

Reply:
Maybe I misunderstood your meaning, I add *#[derive(Serialize, Deserialize)]* for *pub enum SegmentInfo*, and we will use from_slice and to_bytes.